### PR TITLE
Rebase 1.12.4

### PR DESCRIPTION
**Pull request by soltysh**
This is 1.12 rebase, but I think we'll merge that once we are closer to the end. 

/assign @deads2k

### CODE REVIEW

File: pkg/operator/v1alpha1helpers/helper_test.go [hunk 1]
```diff
@@ -278,8 +278,7 @@ func newUnstructured(apiVersion, kind, name, imagePullSpec string) *unstructured
 			"apiVersion": apiVersion,
 			"kind":       kind,
 			"metadata": map[string]interface{}{
-				"name":      name,
-				"namespace": "",
```
[the remark this answers is not in the document]
Comment (soltysh) [reply pkg/operator/v1alpha1helpers/helper_test.go]: With the new rebase, non-namespaced resources don't have namespace set at all, not even empty string. That's why we need to update these.

File: pkg/operator/configobserver/config_observer_controller.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func (c ConfigObserver) sync() error {
 
 	mergedObservedConfig := map[string]interface{}{}
 	for _, observedConfig := range observedConfigs {
-		if err := mergo.Merge(&mergedObservedConfig, observedConfig); err != nil {
+		if err := mergo.Merge(&mergedObservedConfig, observedConfig, mergo.WithOverride); err != nil {
```
[the remark this answers is not in the document]
Comment (sanchezl) [reply pkg/operator/configobserver/config_observer_controller.go]: @soltysh I propose we leave this as is for now (in `monitoring.go`). I don't think its worth the effort to come up with something better at this moment in time.

[the remark this answers is not in the document]
Comment (soltysh) [reply pkg/operator/configobserver/config_observer_controller.go]: @sanchezl fyi, in the current update (fresh rebase as of yesterday) `mergo.WithOverride` is not needed in none except for that one tiny test you've added which checks version of mergo, which probably should be just dropped. Even in in here it's not needed anymore, due to your current test. Expected?

[the remark this answers is not in the document]
Comment (sanchezl) [reply pkg/operator/configobserver/config_observer_controller.go]: @soltysh correct, just drop the tiny test.

**Pull request merged**